src/tasks/node/NodeTaskHelper.ts

```diff
@@ -0,0 +1,212 @@
+        }
+
+        return buildOptions;
+    }
+
+    public async resolveDockerRunOptions(context: DockerRunTaskContext, runDefinition: NodeRunTaskDefinition): Promise<DockerRunOptions> {
+        const helperOptions = runDefinition.node || {};
+        const runOptions = runDefinition.dockerRun;
+
+        const packagePath = NodeTaskHelper.inferPackagePath(helperOptions && helperOptions.package, context.folder);
+        const packageName = await NodeTaskHelper.inferPackageName(packagePath);
+
+        if (runOptions.containerName === undefined) {
+            runOptions.containerName = getDefaultContainerName(packageName);
+        }
+
+        if (runOptions.image === undefined) {
+            runOptions.image = inferImageName(runDefinition, context, packageName);
+        }
+
+        if (helperOptions && helperOptions.enableDebugging) {
+            const inspectMode = helperOptions.inspectMode || 'default';
+            const inspectPort = helperOptions.inspectPort !== undefined ? helperOptions.inspectPort : 9229;
+
+            if (runOptions.command === undefined) {
+                runOptions.command = await NodeTaskHelper.inferCommand(packagePath, inspectMode, inspectPort);
+            }
+
+            if (runOptions.ports === undefined) {
+                runOptions.ports = [];
+            }
+
+            let inspectPortPublished = false;
+
+            // If not already defined, create a mapping for the inspect port between container and host...
+            if (runOptions.ports.find(port => port.containerPort === inspectPort) === undefined) {
+                runOptions.ports.push({
+                    containerPort: inspectPort,
+                    // TODO: Can this mapping be dynamic?
+                    hostPort: inspectPort
+                });
+
+                inspectPortPublished = true;
+            }
+
+            // NOTE: By default, if no ports are explicitly published and the options do not say otherwise, all ports will be published.
+            //       If we published the inspection port, and it was the only published port, that "publish all" behavior would unintentionally be disabled.
+            //       Hence, in that situation, we force "publish all" in addition to the inspection port.
+            if (runOptions.portsPublishAll === undefined && inspectPortPublished && runOptions.ports.length === 1) {
+                runOptions.portsPublishAll = true;
+            }
+        }
+
+        return runOptions;
+    }
+
+    public static inferPackagePath(packageFile: string | undefined, folder: WorkspaceFolder): string {
+        if (packageFile !== undefined) {
+            return resolveFilePath(packageFile, folder);
+        } else {
+            return path.join(folder.uri.fsPath, 'package.json');
+        }
+    }
+
+    public static async inferPackageName(packagePath: string): Promise<string> {
+        const packageJson = await fse.readFile(packagePath, 'utf8');

```

Comment:
Use `fse.readJSON` instead of `fse.readFile` and `JSON.parse`. In general it's more robust and handles some weird file encodings and whatnot

Reply:
Resolved in: https://github.com/microsoft/vscode-docker/pull/1278